scripts/check-swift-format.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+set -euo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+excluded_files=""
+
+if [ -f .swiftformatignore ]; then
+    log "Found swiftformatignore file..."
+    excluded_files=$(cat .swiftformatignore | xargs -I% printf ":(exclude)% ")
+fi
+
+log "Running swift format format..."
+git ls-files -z '*.swift' $excluded_files | xargs -0 swift format format --parallel --in-place
+
+log "Running swift format lint..."
+
+git ls-files -z '*.swift' $excluded_files | xargs -0 swift format lint --strict --parallel
```

Comment:
This is not how you do that in shell (will break with spaces in the file names)... The correct syntax is

```
files=( foo 'foo bar' buz )
echo "${files[@]}"
```

will do what you want

Reply:
Maybe it would have made more sense to use `git ls-files -z ...` and then use `grep -f` to make use of the exclude file?

Or `git ls-files $(cat IGNOREFILE | xargs -I% printf ":(exclude)% ")`

Reply:
> git ls-files $(cat IGNOREFILE | xargs -I% printf ":(exclude)% ")

This has two quoting errors, one in bash and one with xargs. You may be able to leverage `printf "...%q"` which outputs it quoted. But I think the right solution here is


```
#        -X <file>, --exclude-from=<file>
#           Read exclude patterns from <file>; 1 per line.

git ls-files -X IGNOREFILE
```

Reply:
Yeah, doesn't work as advertised :) Only applies to files in the index, which is a bit of a bummer.

Reply:
🤯 infuriating!